src/hooks/API.tsx

```diff
@@ -1,34 +1,38 @@
 import { useEffect, useState } from 'react';
-import { bobaList, toppingsList, milkList } from 'assets/sampleBobaAPI';
+import { ApolloClient, InMemoryCache } from '@apollo/client';
+import { GET_BOBA_LIST } from 'hooks/QueryGraphQL';
 import { ErrorProps } from 'types/common/main';
+import { bobaList, toppingsList, milkList } from 'assets/sampleBobaAPI';
 
 const useApi = (url: string, endpoint: string) => {
   // eslint-disable-next-line  @typescript-eslint/no-explicit-any
   const [data, setData] = useState<any>();
   const [error, setError] = useState<ErrorProps>({ status: 0, message: '' });
   const [loading, setLoading] = useState<boolean>(true);
-  const fetchApi = () => {
-    const fullurl = url + endpoint;
+
+  useEffect(() => {
     if (process.env.NODE_ENV === 'development') {
-      fetch(fullurl)
-        .then((response) => {
-          if (response.ok) {
-            return response.json();
-          } else if (response == undefined) {
-            throw { status: 503, message: 'You forget to turn on backend server...tsk tsk' };
-          } else {
-            throw { status: response.status, message: response.statusText };
-          }
-        })
-        .then((json) => {
-          setData(json);
-        })
-        .catch((err) => {
-          setError(err);
-        })
-        .finally(() => {
+      const fullurl = url + endpoint;
+      const client = new ApolloClient({
+        uri: fullurl,
+        cache: new InMemoryCache(),
+      });
+
+      const fetchData = async () => {
+        try {
+          const { loading, data } = await client.query({
+            query: GET_BOBA_LIST,
```

Comment:
Since this is a reusable `useAPI` hooks, don't hard code `query: GET_BOBA_LIST,`
Take query as gql param for useAPI.
This custom hooks don't need to import or have concept about `import { GET_BOBA_LIST } from 'hooks/QueryGraphQL';`

Reply:
can you explain this abit more? i get that i cant do hard code with GET_BOBA_LIST so do i pass it as props in useAPi ? ( so it will be 2 params to 3 params?

Reply:
There's nothing wrong with having 3 params or accomodating more params to make your `useAPI` more extensible for other.
if you just mentioned `query: GET_BOBA_LIST` what are you gonna do with `GET_TOPPING_LIST ` GET_MILK_LIST`, user or order, etc?
```
const useApi = (url: string, endpoint: string, whatever_query: ANY_SCHEMA) => {
....
// pass that into query
  const { loading, data } = await client.query({
            query: whatever_query
```